From: "Matteo F. Vescovi" <m.vescovi@revese.it>
Date: Wed, 2 Nov 2011 21:16:07 +0100
Subject: Patch to close an old bug about mab2vcard

Thunderbird is now disappeared from Debian archives, replaced by Icedove
(the re-branded version of Thunderbird cleaned from proprietary stuff).
This patch aims to provide full support to conversion of address-book
entries to vCard files, as it was before with Thunderbird.

Closes: #375476
Git-Dch: Short
---
 scripts/mab2vcard |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/scripts/mab2vcard b/scripts/mab2vcard
index d359995..45dfc87 100755
--- a/scripts/mab2vcard
+++ b/scripts/mab2vcard
@@ -131,7 +131,7 @@
 require 5;
 #use diagnostics;
 use strict;
-use constant ABOOK_PATHS => qw (~/.thunderbird/);
+use constant ABOOK_PATHS => qw (~/.icedove/);
 
 #use Data::Dumper;
 my $progname = $0; $progname =~ s@.*/@@g;
@@ -488,14 +488,14 @@ sub escape_results {
 sub locate_default_abook {
 	my @files;
 	foreach my $path (ABOOK_PATHS) {
-		my $dir = glob($path);
-	use Data::Dumper; print Dumper($dir);	
-		next unless $dir;
+		# works around the weird semantics of glob() in scalar mode...
+		my $dir = (glob($path))[0];
+		next unless $dir and -e $dir;
 		error ("$path is not a directory. Please correct ABOOK_PATHS in the script '$progname'")
-			unless -d $dir;
+			unless -d _;
 		$dir=~s/\/*\s*$//;
 print Dumper($dir);
-		push @files,(glob $dir.'/abook.mab');
+		push @files,$dir.'/abook.mab' if -e $dir.'/abook.mab';
 		push @files,(glob $dir.'/*/abook.mab');
 		push @files,(glob $dir.'/*/*/abook.mab');
 	}
-- 
